patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -55,7 +55,29 @@
  
          if (this.field_70170_p.func_180495_p(p_180429_1_.func_177984_a()).func_177230_c() == Blocks.field_150431_aC)
          {
-@@ -1277,12 +1289,12 @@
+@@ -1210,7 +1222,8 @@
+         {
+             this.field_70171_ac = false;
+         }
+-
++        
++        this.field_70171_ac = net.minecraftforge.common.ForgeHooks.isInLiquid(this.func_174813_aQ().func_72314_b(0.0D, -0.4000000059604645D, 0.0D).func_186664_h(0.001D), this); if (field_70171_ac) this.func_70066_B();
+         return this.field_70171_ac;
+     }
+ 
+@@ -1265,7 +1278,10 @@
+         }
+     }
+ 
+-    public boolean func_70055_a(Material p_70055_1_)
++    /**
++     * Checks if the current block the entity is within of the specified material type
++     */
++    @Deprecated public boolean func_70055_a(Material p_70055_1_)
```

Comment:
To minimize the number of changes in this patch, the `@Deprecated` should go on the line before the vanilla method.

The javadoc should be removed, and replaced with a comment after `@Deprecated` which explains what modders should use instead of this method.

Reply:
Okay, will fix in next commit.